src/Microsoft.Azure.WebJobs.Host.Storage/StorageClientProvider.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            _componentFactory = componentFactory;
+            _logForwarder = logForwarder;
+            _logger = logger;
+
+            _logForwarder.Start();
+        }
+
+        /// <summary>
+        /// Gets the subdomain for the resource (i.e. blob, queue, file, table)
+        /// </summary>
+#pragma warning disable CA1056 // URI-like properties should not be strings
+        protected abstract string ServiceUriSubDomain { get; }
+#pragma warning restore CA1056 // URI-like properties should not be strings
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <param name="configuration">Configuration to use</param>
+        /// <param name="client">Client that was created if the connection was valid; otherwise, the default value for the type of <paramref name="client"/>.</param>
+        /// <returns>true if the client could be created; false otherwise</returns>
+        public virtual bool TryGet(string name, IConfiguration configuration, out TClient client)
+        {
+            try
+            {
+                client = Get(name, configuration);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                client = default(TClient);
+                _logger.LogDebug("Could not create Storage Client. Exception: {0}", ex.ToString());
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <param name="resolver">A resolver to interpret the provided connection <paramref name="name"/>.</param>
+        /// <param name="configuration">Configuration to use</param>
+        /// <returns>Client that was created.</returns>
+        public virtual TClient Get(string name, INameResolver resolver, IConfiguration configuration)
+        {
+            var resolvedName = resolver.ResolveWholeString(name);
+            return this.Get(resolvedName, configuration);
+        }
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <param name="configuration">Configuration to use</param>
+        /// <returns>Client that was created.</returns>
+        public virtual TClient Get(string name, IConfiguration configuration)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                name = ConnectionStringNames.Storage; // default
+            }
+
+            // $$$ Where does validation happen?
```

Comment:
I'd say just remove this old comment. Assuming if the connection string is invalid, we'll fail when we attempt to create the client?

Reply:
Correct - it would fail to create the client (and throw an exception).